File: Maths_And_Stats/Geometry/Vector_Product/Vector_Product.py
"""
LANGUAGE: PYTHON
AUTHOR: Diadochokinetic
GITHUB: https://github.com/Diadochokinetic
"""

def Vector_Product(a, b):
    return [a[1]*b[2] - a[2]*b[1], a[2]*b[0] - a[0]*b[2], a[0]*b[1] - a[1]*b[0]]

def float_input(string_input):
    return [float(element) for element in string_input.split()]

if __name__ == '__main__':
    a = float_input(input("Please enter a vector containing 3 elements, e.g. 1 2 3: "))
    b = float_input(input("Please enter another vector containing 3 elements, e.g. 4 5 6: "))
    print(f'The vector product for {a} and {b} is: {Vector_Product(a, b)}')

File: Maths/Geometry/Vector_Product/Vector_Product.hs
module Vector_Product (vector) where

[x1,x2,x3] `vector` [y1,y2,y3] = 
    [x2 * y3 - x3 * y2,
     x3 * y1 - x1 * y3,
     x1 * y2 - x2 * y1]
